src/cryptonote_core/blockchain.cpp

```diff
@@ -3877,11 +3877,13 @@ bool Blockchain::cleanup_handle_incoming_blocks(bool force_sync)
         store_blockchain();
       m_sync_counter = 0;
     }
-    else if (m_db_blocks_per_sync && m_sync_counter >= m_db_blocks_per_sync)
+    else if ((m_db_sync_on_blocks && m_sync_counter >= m_db_sync_threshold) || (!m_db_sync_on_blocks && m_bytes_to_sync >= m_db_sync_threshold))
```

Comment:
Since `m_db_blocks_per_sync` is now named as `m_db_sync_threshold `, shouldn't this `if` statement also check for `m_db_sync_threshold  && (...)` in order to preserve the logic?